### PR TITLE
Fixes #1372: Add localization plugin documentation

### LINKED ISSUE #1372: Add documentation for localization plugin

### PR DESCRIPTION
**Pull request by yiqunw700**
- Add the localization doc page
- Update the figma-features page

### CODE REVIEW

File: docs/_docs/localization.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+---
+title: Localization
+layout: page
+nav_order: 12
+---
+
+# Localization
+
+We are developing tools for developers to localize the figma design into different languages more easily.
+The first introduced tool is a figma plugin that generates a strings xml files containing the strings
```
Comment (timothyfroehlich) [inline docs/_docs/localization.md:10]:
```suggestion
The first introduced tool is a figma plugin that generates a strings xml file containing the strings
```

**Review by timothyfroehlich — APPROVED** [review]
This looks great!

File: docs/_docs/localization.md (hunk 1 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done. Good catch!

File: support-figma/extended-layout-plugin/README.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ This plugin adds some additional capabilities to Figma specifically for Automoti
 - Check document for errors: Using a JSON file created from the application's DesignCompose annotations, check for possible errors in the document such as missing or incorrectly named nodes
 - Check/update keywords: Upload a new JSON file created from the application's DesignCompose annotations
 - Sync Prototype changes: Copies the interactive fields from the JS API to a data area that the REST API can fetch (since our webservice can't talk to the JS API). This needs to be run anytime an interaction changes.
+- Localization: Assigns text node a string resource name and generates a strings xml file to integrate with translations for the Android app.
```
Comment (rylin8) [inline support-figma/extended-layout-plugin/README.md:12]: "text nodes" should be plural.
```suggestion
- Localization: Assigns text nodes a string resource name and generates a strings xml file to integrate with translations for the Android app.
```

File: docs/_docs/localization.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+# Localization
+
+We are developing tools for developers to localize the figma design into different languages more easily.
+The first introduced tool is a figma plugin that generates a strings xml file containing the strings
+used by the figma design.
+
+## 1. Build the localization plugin
+
+Localization plugin is still under development and it is only available from building the source code.
```
Comment (rylin8) [inline docs/_docs/localization.md:15]:
```suggestion
The localization plugin is still under development and is only available from building the source code.
```

File: docs/_docs/localization.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+The first option excludes text nodes whose name starts with `#` which is a naming convention for text
+nodes designed for app customization.
+
+To use the second option, you will need to run the **Check/update keywords** plugin first to upload
+the generated json file under your android app's build folder. The generated json file contains app
+customizations for nodes so the localization plugin can exclude those text nodes from translation.
+
+<img src="CheckOrUpdateKeywords.png">
+
+### 2.2 Upload an strings xml
```
Comment (rylin8) [inline docs/_docs/localization.md:67]:
```suggestion
### 2.2 Upload a strings xml file
```

File: docs/_docs/localization.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+### 3.3 Live update option
+
+If you use live update, because DesignCompose uses the local string from the resource file, you will
+not be able to see the changes you are making to the text nodes in live update. We have provided an option
+to re-enable the live update for text nodes by unchecking the `Use local string resource if exists` option.
+
+<img src="LiveUpdateLocalizationOption.png">
+
+### 3.5 Localize app side customizations
```
Comment (rylin8) [inline docs/_docs/localization.md:123]: You have 3.5 and 3.4 reversed

File: docs/_docs/localization.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+For a text node with a single style, the plugin generates a string resource. And for a text node with
+mixed styles, the plugin generates a string array resource with each item mapping to the text segments
+with different styles.
+
+## 5. What is next
+
+We are working on a few improvements for the plugin:
+
+1. the ability to exclude a text node manually before hitting the **Generate** button. Designers might
```
Comment (rylin8) [inline docs/_docs/localization.md:187]:
```suggestion
1. The ability to exclude a text node manually before hitting the **Generate** button. Designers might
```

File: docs/_docs/localization.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+add some text nodes as comments or explanations to design components which are not actually part of
+the product design.
+2. the ability to do different translations for the same text. Right now the plugin groups the text nodes
```
Comment (rylin8) [inline docs/_docs/localization.md:190]:
```suggestion
2. The ability to do different translations for the same text. Right now the plugin groups the text nodes
```

File: docs/_docs/localization.md [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+with the same text to use the same string resource name. However, same text can mean differently under
```
Comment (rylin8) [inline docs/_docs/localization.md:191]:
```suggestion
with the same text to use the same string resource name. However, the same text can have different translations under
```

File: docs/_docs/localization.md [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+different context.
```
Comment (rylin8) [inline docs/_docs/localization.md:192]:
```suggestion
different contexts.
```

File: docs/_docs/localization.md [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+
+## 6. Summary
+
+This plugin can not be 100% perfect to solve the localization difficulties, but we will
```
Comment (rylin8) [inline docs/_docs/localization.md:196]:
```suggestion
This plugin does not perfectly solve all localization issues, but we will
```

File: docs/_docs/localization.md [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+continue to improve it to make localization easier for DesignCompose developers. Any suggestion
```
Comment (rylin8) [inline docs/_docs/localization.md:197]:
```suggestion
continue to improve it to make localization easier for DesignCompose developers. Any suggestions
```

**Review by rylin8 — COMMENTED** [review]
Looks great, I just added a lot of minor wording suggestions.

File: support-figma/extended-layout-plugin/README.md (hunk 2 above)
Comment (yiqunw700) [reply support-figma/extended-layout-plugin/README.md]: Done.

File: docs/_docs/localization.md (hunk 3 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 4 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 6 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 7 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 8 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 9 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 10 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 11 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

File: docs/_docs/localization.md (hunk 5 above)
Comment (yiqunw700) [reply docs/_docs/localization.md]: Done.

**Review by rylin8 — APPROVED** [review]

**Pull request merged**